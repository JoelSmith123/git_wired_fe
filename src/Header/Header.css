.Header {
  background-color: #4c4c4c;
  display: flex;
  justify-content: space-between;
  height: 100px;
}

.header-btn {
  background: transparent; 
  border: none;
  color: #6666ff;
  font-size: 1rem;
  height: 100%;
  vertical-align: middle;
  width: 100px;
}

.header-recent-projects-btn-dropdown-container {
  display: inline-block;
  height: 100%;
  vertical-align: middle;
  width: 100px;
}

.header-recent-projects-btn-dropdown {
  position: relative;
  z-index: 1;
}

.header-recent-projects-btn-dropdown-btn {
  height: 40px;
  width: 150px;
}

.header-home-btn {
  border-right: 1px solid #fff;
}

.header-help-btn {
  border-right: 1px solid #fff;
}

.header-recent-projects-btn {
  border-left: 1px solid #fff;
  border-right: 1px solid #fff;  
}

.header-login-btn {
  border-left: 1px solid #fff;
}

.header-register-btn {
  border-left: 1px solid #fff;
}

.header-logout-btn {
  border-left: 1px solid #fff;
}

.header-title-container {
  text-align: center;
}

.header-main-title {
  color: #fff;
  margin-top: 0;
  margin-bottom: 0;
  padding-top: 15px;
}

.header-sub-title {
  color: #fff;
  margin-top: 5px;
}

