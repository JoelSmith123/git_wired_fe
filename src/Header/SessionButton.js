
import React, { Component } from 'react';

export default class Header extends Component {

  constructor(props) {
    super(props)
    this.updateState = this.props.changeLoggedInState.bind(this)
  }


 goToAuth = () => {
   window.location.href='https://git-wired-be.herokuapp.com/authenticate'
 }

 loginButton = () => {
   return (
     <input type="button"
            onClick={ () => {
              this.props.user.setGitWiredToken('123');
              this.updateState()
                // this.goToAuth()
              }
            }
            className='header-btn header-login-btn'
            value="Login"
    />
   )
 }

 logoutButton = () => {
   return (
     <input type="button"
            onClick={ () => {
              this.props.user.logout();
              this.updateState()
              }
            }
            className='header-btn header-logout-btn'
            value="Logout"
    />
   )
 }

 render() {
  return (
    <React.Fragment>
      { this.props.user.isLoggedIn() ? this.logoutButton() : this.loginButton() }
    </React.Fragment>
  )
 }

}
