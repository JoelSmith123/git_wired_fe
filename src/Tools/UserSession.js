export default class UserSession {
  constructor() {
    this.tokenKey = 'userToken'
  }

  setGitWiredToken(token) {
    sessionStorage.setItem(this.tokenKey, token);
  }

  getGitWiredToken() {
    return sessionStorage.getItem(this.tokenKey);
  }

  logout() {
    sessionStorage.removeItem(this.tokenKey);
  }


  isLoggedIn = () => {
    return Boolean( this.getGitWiredToken() )
  }




  // Make Fetch /login call
  // receive userToken

  // TO DO - TEST ME
  //  - CONFIRM endpoint
  //  - CONFIRM RESPONSE STRUCTURE
  // loginUser() {
  //   let url = 'https://git-wired-be.herokuapp.com/login'
  //   fetch(url)
  //     .then(response => response.json())
  //     .then(data     => this.getToken(data).bind(this))
  //     .then(token    => this.setGitWiredToken(token).bind(this))
  //     .catch(error   => { console.log(error); });
  // }
  //
  // getToken(data) {
  //   let token = data[this.tokenKey]
  //   return token
  // }


}
