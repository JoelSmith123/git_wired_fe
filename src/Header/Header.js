import React, { Component } from 'react';
import './Header.css';

import SessionButton from './SessionButton.js'

export default class Header extends Component {
  constructor() {
    super()
    this.state = {
      showDropdown: false
    }
  }

  toggleDropdown = (event) => {
    event.preventDefault()

    this.setState({ showDropdown: !this.state.showDropdown })
  }

  handleTemplateSelection = (event, templateName) => {
    event.preventDefault()

    this.setState({ showDropdown: false })
    this.props.selectTemplate(event, templateName)
  }


  render() {
    if (this.props.loggedIn) {
      return (
        <div className='Header'>
          <span className='header-btn-container'>
            <button className='header-btn header-home-btn'
                    onClick = { () => { this.props.changeViewPageState('welcome') } }
            >Git Wired</button>
            <input type="button"
                   onClick={ () => { window.location.href='https://github.com/JoelSmith123/git_wired_fe/blob/master/README.md' } }
                   className='header-btn header-help-btn'
                   value="Help"
            />
          </span>
          <span className='header-btn-container'>
            <div className='header-recent-projects-btn-dropdown-container'>
              <button className='header-btn header-recent-projects-btn'
              // onClick={this.toggleDropdown}
                      onClick = { () => { this.props.changeViewPageState('recent') } }
              >Recent Projects</button>
              {
                this.state.showDropdown ? (
                  <div className='header-recent-projects-btn-dropdown'>
                    <button className='header-recent-projects-btn-dropdown-btn'
                            name='card-page-template'
                            onClick={(event) => this.handleTemplateSelection(event, event.target.name)}
                    >card page template
                    </button>
                    <button className='header-recent-projects-btn-dropdown-btn'
                            name='blog-page-template'
                            onClick={(event) => this.handleTemplateSelection(event, event.target.name)}
                    >blog page template
                    </button>
                    <button className='header-recent-projects-btn-dropdown-btn'
                            name='blog-post-template'
                            onClick={(event) => this.handleTemplateSelection(event, event.target.name)}
                    >blog post template
                    </button>
                  </div>
                ) : (
                  null
                )
              }
            </div>
            <button className='header-btn header-profile-btn'
                    // onClick={this.props.viewProfile}
                    onClick = { () => { this.props.changeViewPageState('profile') } }
            >Profile
            </button>
            <SessionButton user={this.props.user} changeLoggedInState={this.props.changeLoggedInState}/>
          </span>
        </div>
      )
    } else {
      return (
        <div className='Header'>
          <button className='header-btn header-home-btn'>Git Wired</button>
          <span className='header-title-container'>
            <h1 className='header-main-title'>Welcome!</h1>
            <h3 className='header-sub-title'><a>Login</a> to get started!</h3>
          </span>
          <span className='header-btn-container'>
            <SessionButton user={this.props.user} changeLoggedInState={this.props.changeLoggedInState}/>
          </span>
        </div>
      )
    }
  }
}
