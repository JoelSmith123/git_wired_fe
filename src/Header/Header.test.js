import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Header from './Header.js'
import UserSession from '../Tools/UserSession.js'

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header user={ new UserSession } changeLoggedInState={jest.fn()}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Header user={ new UserSession } changeLoggedInState={jest.fn()}/>)
    expect(wrapper).toMatchSnapshot()
  })  

  it('should toggle dropdown state on toggleDropdown invocation', () => {
    const wrapper = shallow(<Header user={ new UserSession }/>)
    const mockEvent = { target: {}, preventDefault: jest.fn() }

    wrapper.setState({ showDropdown: false })
    wrapper.instance().toggleDropdown(mockEvent)

    expect(wrapper.state('showDropdown')).toEqual(true)
  })

  it('should change dropdown state', () => {
    const wrapper = shallow(<Header user={ new UserSession } selectTemplate={jest.fn()}/>)
    const mockEvent = { target: {}, preventDefault: jest.fn() }
    const mockTemplateName = 'template name'

    wrapper.setState({ showDropdown: true })
    wrapper.instance().handleTemplateSelection(mockEvent, mockTemplateName)

    expect(wrapper.state('showDropdown')).toEqual(false)
  })

  it('should conditionally render header based on loggedIn state true', () => {
    const wrapper = mount(<Header user={ new UserSession } selectTemplate={jest.fn()} loggedIn={true} changeLoggedInState={jest.fn()}/>)

    expect(wrapper.find('.header-recent-projects-btn-dropdown-container')).toHaveLength(1)
  })

  it('should conditionally render header based on loggedIn state false', () => {
    const wrapper = mount(<Header user={ new UserSession } selectTemplate={jest.fn()} loggedIn={false} changeLoggedInState={jest.fn()}/>)

    expect(wrapper.find('.header-title-container')).toHaveLength(1)
  }) 

  it('should call handleTemplateSelection on button click', () => {
    const wrapper = mount(<Header user={ new UserSession } selectTemplate={jest.fn()} loggedIn={false} changeLoggedInState={jest.fn()}/>)
    wrapper.setState({ loggedIn: true, showDropdown: true })
    wrapper.find('.header-recent-projects-btn-dropdown-btn').forEach((btn) => {
      btn.simulate('click')
      expect(wrapper.instance().handleTemplateSelection()).toHaveBeenCalled()
    })
  }) 

});
