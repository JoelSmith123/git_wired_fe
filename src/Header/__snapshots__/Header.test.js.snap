// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    changeLoggedInState={[MockFunction]}
    user={
      UserSession {
        "getURLToken": [Function],
        "isLoggedIn": [Function],
        "login": [Function],
        "tokenKey": "userToken",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="header-btn header-home-btn"
        >
          Git Wired
        </button>,
        <span
          className="header-title-container"
        >
          <h1
            className="header-main-title"
          >
            Welcome!
          </h1>
          <h3
            className="header-sub-title"
          >
            <a>
              Login
            </a>
             to get started!
          </h3>
        </span>,
        <span
          className="header-btn-container"
        >
          <Header
            changeLoggedInState={[MockFunction]}
            user={
              UserSession {
                "getURLToken": [Function],
                "isLoggedIn": [Function],
                "login": [Function],
                "tokenKey": "userToken",
              }
            }
          />
        </span>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Git Wired",
          "className": "header-btn header-home-btn",
        },
        "ref": null,
        "rendered": "Git Wired",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="header-main-title"
            >
              Welcome!
            </h1>,
            <h3
              className="header-sub-title"
            >
              <a>
                Login
              </a>
               to get started!
            </h3>,
          ],
          "className": "header-title-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome!",
              "className": "header-main-title",
            },
            "ref": null,
            "rendered": "Welcome!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a>
                  Login
                </a>,
                " to get started!",
              ],
              "className": "header-sub-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                },
                "ref": null,
                "rendered": "Login",
                "type": "a",
              },
              " to get started!",
            ],
            "type": "h3",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header
            changeLoggedInState={[MockFunction]}
            user={
              UserSession {
                "getURLToken": [Function],
                "isLoggedIn": [Function],
                "login": [Function],
                "tokenKey": "userToken",
              }
            }
          />,
          "className": "header-btn-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changeLoggedInState": [MockFunction],
            "user": UserSession {
              "getURLToken": [Function],
              "isLoggedIn": [Function],
              "login": [Function],
              "tokenKey": "userToken",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="header-btn header-home-btn"
          >
            Git Wired
          </button>,
          <span
            className="header-title-container"
          >
            <h1
              className="header-main-title"
            >
              Welcome!
            </h1>
            <h3
              className="header-sub-title"
            >
              <a>
                Login
              </a>
               to get started!
            </h3>
          </span>,
          <span
            className="header-btn-container"
          >
            <Header
              changeLoggedInState={[MockFunction]}
              user={
                UserSession {
                  "getURLToken": [Function],
                  "isLoggedIn": [Function],
                  "login": [Function],
                  "tokenKey": "userToken",
                }
              }
            />
          </span>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Git Wired",
            "className": "header-btn header-home-btn",
          },
          "ref": null,
          "rendered": "Git Wired",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="header-main-title"
              >
                Welcome!
              </h1>,
              <h3
                className="header-sub-title"
              >
                <a>
                  Login
                </a>
                 to get started!
              </h3>,
            ],
            "className": "header-title-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome!",
                "className": "header-main-title",
              },
              "ref": null,
              "rendered": "Welcome!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a>
                    Login
                  </a>,
                  " to get started!",
                ],
                "className": "header-sub-title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "a",
                },
                " to get started!",
              ],
              "type": "h3",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              changeLoggedInState={[MockFunction]}
              user={
                UserSession {
                  "getURLToken": [Function],
                  "isLoggedIn": [Function],
                  "login": [Function],
                  "tokenKey": "userToken",
                }
              }
            />,
            "className": "header-btn-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "changeLoggedInState": [MockFunction],
              "user": UserSession {
                "getURLToken": [Function],
                "isLoggedIn": [Function],
                "login": [Function],
                "tokenKey": "userToken",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
